### PR TITLE
Add Payment Splitter Payees Length Getter

### LINKED ISSUE #3818: Add PaymentSplitter.payeesLength()
**🧐 Motivation**

I want to list all payees of a `PaymentSplitter` contract.

**📝 Details**

`PaymentSplitter` has a `.payee(index)` method but it is hard to use it without knowing how many payees are there.

Consider adding this to the `PaymentSplitter` contract:
```solidity
function payeesLength() public view returns (uint256) {
  return _payees.length;
}
```

### PR DESCRIPTION
**Pull request by lhemerly**
Fixes #3818

Add payment splitter payess length getter

- [x] Tests
- [x] Documentation
- [x] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: contracts/finance/PaymentSplitter.sol [hunk 1]
```diff
@@ -184,6 +184,16 @@ contract PaymentSplitter is Context {
         emit ERC20PaymentReleased(token, account, payment);
     }
 
+    /**
+     *
+     * @dev getter for payees length.
+     *
+     */
+
```
Comment (ernestognw) [inline contracts/finance/PaymentSplitter.sol:192]: This is compiled into the documentation so it's important to present it formatted

```suggestion
/**
 * @dev Getter for the amount of payees.
 */
```

File: contracts/finance/PaymentSplitter.sol [hunk 2]
```diff
@@ -184,6 +184,16 @@ contract PaymentSplitter is Context {
...
+    function getPayeesLength() public view returns (uint256) {
```
Comment (ernestognw) [inline contracts/finance/PaymentSplitter.sol:193]: I think it makes sense to call it `totalPayees` or just `payees` to follow the contract convention. I'd wait for comments from @Amxx

File: .changeset/dirty-clocks-smoke.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': minor
+---
+
+Payment Splitter: add payees length getter
```
Comment (ernestognw) [inline .changeset/dirty-clocks-smoke.md:5]:
```suggestion
`PaymentSplitter`: Add `getPayeesLength()` function returning the 
```

**Review by ernestognw — COMMENTED** [review]
Thanks @lhemerly! I'll wait for comments from @Amxx

File: contracts/finance/PaymentSplitter.sol (hunk 2 above)
Comment (Amxx) [reply contracts/finance/PaymentSplitter.sol:190]: `totalPayees()` or `payeeCount()` would either be fine by me.

Comment (Amxx) [reply contracts/finance/PaymentSplitter.sol:190]: Also, this function should move to line 123, just after `payee(uint256)`

File: .changeset/dirty-clocks-smoke.md (hunk 3 above)
Comment (ernestognw) [reply .changeset/dirty-clocks-smoke.md]: Sorry, just realized I didn't finish this suggestion. I'll send a commit with adjustments

File: contracts/finance/PaymentSplitter.sol (hunk 2 above)
Comment (ernestognw) [reply contracts/finance/PaymentSplitter.sol]: I went for `totalPayees` and put the function after `totalReleased` following the order of `total*` getters followed by their `*` counterparts in the same order.

**Review by ernestognw — APPROVED** [review]
This looks good to me. Thanks, @lhemerly.
As per policy, we require 2 people to approve a PR before merging, so you'd expect either @Amxx approval or more follow-up comments (if any).

**Comment by Amxx** [comment]
With 4.9 audit being under way, we don't want to add anything in the master branch that would get in the release without having been audited. Either we merge that into another branch (v4.10?) of we freeze all merges until 4.9 has been released.

File: contracts/finance/PaymentSplitter.sol [hunk 4]
```diff
@@ -92,6 +92,13 @@ contract PaymentSplitter is Context {
         return _erc20TotalReleased[token];
     }
 
+    /**
+     * @dev Getter for the amount of payees.
+     */
+    function totalPayees() public view returns (uint256) {
+        return _payees.length;
+    }
+
```
Comment (Amxx) [inline contracts/finance/PaymentSplitter.sol:101]: Also, as I said before (somehow my comment was missed), this new function should go just after `function payee(uint256)`.

Function ordering is important for readability / auditability

File: contracts/finance/PaymentSplitter.sol (hunk 2 above)
Comment (Amxx) [reply contracts/finance/PaymentSplitter.sol]: it has nothing to do with `totalReleased`. This function works in pair with `function payee(uint256)`, so they should go together.

**Comment by frangio** [comment]
PaymentSplitter has been removed for now in 5.0.

**Pull request closed without merging**